android/app/src/main/java/com/sample/component/ComponentActivity.kt

```diff
@@ -0,0 +1,79 @@
+package com.sample.component
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.MenuItem
+import androidx.fragment.app.Fragment
+import com.facebook.react.ReactActivity
+import com.facebook.soloader.SoLoader
+
+class ComponentActivity : ReactActivity() {
+
+    companion object {
+        private const val COMPONENT_NAME = "extra:componentName"
+        private const val COMPONENT_DISPLAY_NAME = "extra:componentDisplayName"
+
+        fun newIntent(
+            activity: Activity,
+            componentName: String,
+            componentDisplayName: String
+        ): Intent {
+            return Intent(activity, ComponentActivity::class.java).apply {
+                putExtra(COMPONENT_NAME, componentName)
+                putExtra(COMPONENT_DISPLAY_NAME, componentDisplayName)
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        SoLoader.init(this, false)
+
+        val componentDisplayName = intent.extras?.getString(COMPONENT_DISPLAY_NAME, null)
+            ?: throw IllegalArgumentException("Component display name has to be provided.")
```

Comment:
Can't we just fallback to `componentName`? I don't think it should be required.

Reply:
Is it? If `displayName` is optional, I will have to change the manifest types as well to make the field optional.

Reply:
_One could also forgo the name, in which case the `entryPoint` string will be used instead._

-- https://github.com/tido64/discussions-and-proposals/blob/tido/test-app-as-package/proposals/0000-React-Native-Test-App-as-a-Package.md#the-manifest